java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -339,6 +339,59 @@ private static void dumpConfig(final ConfigContext context) {
         }
     }
 
+
+    /* ======================================================================
+     * MantaClient content-type Utility Methods
+     * ====================================================================== */
+
+    /**
+     * Depending on the configuration value to enable/disable automatic content-type detection
+     * within the Client SDK {@link ConfigContext#isContentTypeDetectionEnabled()}, assign
+     * content-type value accordingly.
+     *
+     * @param headers optional HTTP headers to include when copying the object.
+     * @param path The formatted path to the Manta object.
+     * @param file file to upload in Manta using the SDK.
+     * @return content type object
+     * @throws IOException when there is a problem sending the object over the network
+     */
+    private ContentType assignContentType(final MantaHttpHeaders headers,
+                                          final String path,
+                                          final File file) throws IOException {
+        if (BooleanUtils.isTrue(config.isContentTypeDetectionEnabled())) {
+            return ContentTypeLookup.findOrDefaultContentType(headers,
+                    path,
+                    file,
+                    ContentType.APPLICATION_OCTET_STREAM);
+        } else {
+            return ContentType.APPLICATION_OCTET_STREAM;
+        }
+    }
+
+    /**
+     * Depending on the configuration value to enable/disable automatic content-type detection
+     * within the Client SDK {@link ConfigContext#isContentTypeDetectionEnabled()}, assign
+     * content-type value accordingly.
+     *
+     * @param headers optional HTTP headers to include when copying the object.
+     * @param path The formatted path to the Manta object.
+     * @return content type object
+     */
+    @SuppressWarnings("Duplicates")
```

Comment:
Is this still needed?

Reply:
Yes, since the following method has the same duplicated code block and its showing up as a ```warning``` in ```IntelliJ```